barretenberg/cpp/src/barretenberg/crypto/merkle_tree/indexed_tree/content_addressed_indexed_tree.hpp

```diff
@@ -1422,6 +1424,14 @@ void ContentAddressedIndexedTree<Store, HashingPolicy>::add_or_update_values_seq
     const AddSequentiallyCompletionCallbackWithWitness& completion,
     bool capture_witness)
 {
+
+    // This struct is used to collect some state from the asynchronous operations we are about to perform
+    struct IntermediateResults {
+        std::shared_ptr<std::vector<InsertionUpdates>> updates_to_perform;
+        size_t appended_leaves = 0;
+    };
+    std::shared_ptr<IntermediateResults> results = std::make_shared<IntermediateResults>();
```

Comment:
Usual idiom when creating smart pointers is

`auto var = std::make_xxx<Type>(...);`

No need to repeat the type more than once. The rule in general is "use auto if and only if the type is clear from context".